### PR TITLE
Rename the shell function to 'vcpkg-shell'.

### PR DESCRIPTION
**Pull request by BillyONeal**
Resolves https://github.com/microsoft/vcpkg/issues/32641 for good.

This depends on negotiation with vcpkg-artifacts stakeholders et al.

Note that the shell function is only necessary in cases the user wants their shell messed with:

image

### CODE REVIEW

File: azure-pipelines/end-to-end-tests-dir/artifacts.ps1 [hunk 1]
```diff
@@ -210,7 +210,7 @@ try {
     #  can't deactivate without postscript:
     $output = Run-VcpkgAndCaptureOutput -ForceExe deactivate
     Throw-IfNotFailed
-    Test-Match $output "no postscript file: rerun with the vcpkg shell function rather than executable"
+    Test-Match $output "no postscript file: rerun with the same arguments as vcpkg-shell"
```
Comment (ras0219-msft) [inline azure-pipelines/end-to-end-tests-dir/artifacts.ps1:213]:
```suggestion
    Test-Match $output "no postscript file: run vcpkg-shell with the same arguments"
```
Consider this wording; no change requested if you don't think it's more understandable.

File: vcpkg-init/vcpkg-init [hunk 2]
```diff
@@ -82,7 +82,7 @@ if [ $? -eq 1 ]; then
 fi
 
 # So, we're the real script then.
-vcpkg() {
+vcpkg-shell() {
```
Comment (ras0219-msft) [inline vcpkg-init/vcpkg-init:85]: This should also be renamed in the comment at the top of the file:

> can use the vcpkg() function.

File: vcpkg-init/vcpkg-init.ps1 [hunk 3]
```diff
@@ -80,7 +80,7 @@ if(-Not (bootstrap-vcpkg)) {
 # Export vcpkg to the current shell.
 New-Module -name vcpkg -ArgumentList @($VCPKG) -ScriptBlock {
   param($VCPKG)
-  function vcpkg() {
+  function vcpkg-shell() {
```
Comment (ras0219-msft) [inline vcpkg-init/vcpkg-init.ps1:83]: This should be renamed in the top comment of this file:

> can use the vcpkg() function

File: vcpkg-init/vcpkg-init.ps1 [hunk 4]
```diff
@@ -92,8 +92,10 @@ New-Module -name vcpkg -ArgumentList @($VCPKG) -ScriptBlock {
         iex $postscr
       }
 
-      Remove-Item -Force -ea 0 $env:Z_VCPKG_POSTSCRIPT,env:Z_VCPKG_POSTSCRIPT
+      Remove-Item -Force -ea 0 $env:Z_VCPKG_POSTSCRIPT
     }
+
+    Remove-Item env:Z_VCPKG_POSTSCRIPT
   }
 } | Out-Null
```
Comment (ras0219-msft) [inline vcpkg-init/vcpkg-init.ps1:101]: On line 103 (two below this), shouldn't it call `return vcpkg-shell @args`?

**Review by ras0219-msft — APPROVED** [review]

File: azure-pipelines/end-to-end-tests-dir/artifacts.ps1 (hunk 1 above)
Comment (BillyONeal) [reply azure-pipelines/end-to-end-tests-dir/artifacts.ps1:213]: I think your suggestion is an improvement

**Pull request merged**